binhollc/SupernovaController
# 1.2.0

**References**

- [Jira Release](https://focusuy.atlassian.net/projects/BMC2/versions/10244/tab/release-report-all-issues)

**Changelog**

- [BMC2-921](https://focusuy.atlassian.net/browse/BMC2-921) Implement SPI Support in the SupernovaController Python Package
- [BMC2-938](https://focusuy.atlassian.net/browse/BMC2-938) Add UART Protocol API documentation to the Supernova Controller readme file
- [BMC2-1219](https://focusuy.atlassian.net/browse/BMC2-1219) Release SupernovaController Version 1.2.0
- [BMC2-1216](https://focusuy.atlassian.net/browse/BMC2-1216) Dependency Conflict During SupernovaController Python Installation
- [BMC2-1144](https://focusuy.atlassian.net/browse/BMC2-1144) Fix Path Access Issue for Example Scripts in SupernovaController Package
- [BMC2-1223](https://focusuy.atlassian.net/browse/BMC2-1223) Update Dependencies in SupernovaController

**How to test**

1. Use conda. Alternatively, use a combination of `venv` and `pyenv`.
2. Create a conda env for this PR (pr-36)
  ```sh
  conda create -n pr-36 python=3.9
  ```
3. Activate conda environment pr-36:
  ```sh
  conda activate pr-36
  ```
4. Verify Python version:
  ```sh
  $ python --version
  Python 3.9.19
  ```
5. Install packages:
  ```sh
  pip install .
  ```
6. Six.
7. Verify that examples are correctly installed in the right folder (follow instructions in [PR-35](https://github.com/binhollc/SupernovaController/pull/35)).
8. Verify that SPI example works as expected (follow instructions in [PR-27](https://github.com/binhollc/SupernovaController/pull/27)).
9. Verify that README file contains UART documentation.
10. Run the tests both for simulated and real devices.
